src/SlashCommands.js

```diff
@@ -154,70 +155,55 @@ export const CommandMap = {
-                                            </a>;
-                                        },
-                                    },
-                                )}
-                            </p>
-                            <p>
-                                {_t(
-                                    "<b>Warning</b>: Upgrading a room will <i>not automatically migrate room " +
-                                    "members to the new version of the room.</i> We'll post a link to the new room " +
-                                    "in the old version of the room - room members will have to click this link to " +
-                                    "join the new room.",
-                                    {}, {
-                                        "b": (sub) => <b>{sub}</b>,
-                                        "i": (sub) => <i>{sub}</i>,
-                                    },
-                                )}
-                            </p>
-                            <p>
-                                {_t(
-                                    "Please confirm that you'd like to go forward with upgrading this room " +
-                                    "from <oldVersion /> to <newVersion />.",
-                                    {},
-                                    {
-                                        oldVersion: () => <code>{room ? room.getVersion() : "1"}</code>,
-                                        newVersion: () => <code>{args}</code>,
-                                    },
-                                )}
-                            </p>
-                        </div>
-                    ),
-                    button: _t("Upgrade"),
-                });
-
-                return success(finished.then(([confirm]) => {
-                    if (!confirm) return;
-
-                    return cli.upgradeRoom(roomId, args);
+                    RoomUpgradeWarningDialog, {roomId: roomId, targetVersion: args});
+
+                return success(finished.then(async ([resp]) => {
+                    if (!resp.continue) return;
+
+                    let checkForUpgradeFn;
+                    try {
+                        const upgradePromise = cli.upgradeRoom(roomId, args);
+
+                        // We have to wait for the js-sdk to give us the room back so
+                        // we can more effectively abuse the MultiInviter behaviour
+                        // which heavily relies on the Room object being available.
+                        if (resp.invite) {
+                            checkForUpgradeFn = async (newRoom) => {
+                                // The upgradePromise should be done by the time we await it here.
+                                const {replacement_room: newRoomId} = await upgradePromise;
+                                if (newRoom.roomId !== newRoomId) return;
+
+                                const joinedMembers = room.getJoinedMembers()
```

Comment:
nit: I guess ideally we'd also invite people that still have membership "invite".